### ISSUE TITLE
Cache flag ignoring .eslintignore changes

### ISSUE THREAD

**Issue by galvarez421** [eslint/eslint#11316]
**Tell us about your environment**

* **ESLint Version:** 5.12.1
* **Node Version:** 10.4.1
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Steps to reproduce using test case repository/code**
1. Clone https://github.com/galvarez421/eslint-cache-ignore-test.
2. Run `npm install`.
3. Run `npm run test-cache-ignore`.
4. See ESLint warning `File ignored because of a matching ignore pattern. Use "--no-ignore" to override`.
5. Erase `test.js` from .eslintignore and save the file.
6. Run `npm run test-cache-ignore` again.
7. Still see ESLint warning `File ignored because of a matching ignore pattern. Use "--no-ignore" to override`.

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

The JS file being linted is empty. See https://github.com/galvarez421/eslint-cache-ignore-test.

From package.json "test-cache-ignore" script:
```bash
eslint test.js --cache
```

From Git Bash terminal:
```bash
npm run test-cache-ignore
```

**What did you expect to happen?**
I expected that after I changed the .eslintignore file, running the npm script again would not display the warning message about the file being ignored. It seems that the only way to prevent that warning error from appearing after making a change to .eslintignore is to remove the `--cache` flag.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ npm run test-cache-ignore

> eslint-cache-ignore-test@1.0.0 test-cache-ignore C:\repos\eslint-cache-ignore-test
> eslint test.js --cache


C:\repos\eslint-cache-ignore-test\test.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

✖ 1 problem (0 errors, 1 warning)
```

**Are you willing to submit a pull request to fix this bug?**
Not sure. Depends on if this is expected behavior...
